airbyte-cdk/python/airbyte_cdk/sources/declarative/yaml_declarative_source.py

```diff
@@ -79,3 +82,74 @@ def _stream_configs(self):
+        with respect to interfaces that are contained within generic data types.
+        :param expand_class: The declarative component class that will have its interface fields expanded
+        :param visited: cache used to store a record of already visited declarative classes that have already been seen
+        :return: The expanded declarative component
+        """
+
+        # Recursive base case to stop recursion if we have already expanded an interface in case of cyclical components
+        # like CompositeErrorHandler
+        if expand_class.__name__ in visited:
+            return visited[expand_class.__name__]
+        visited[expand_class.__name__] = expand_class
+
+        next_classes = []
+        copy_cls = type(expand_class.__name__, expand_class.__bases__, dict(expand_class.__dict__))
+        class_fields = fields(copy_cls)
+        for field in class_fields:
+            unpacked_field_types = DeclarativeComponentFactory.unpack(field.type)
+            copy_cls.__annotations__[field.name] = unpacked_field_types
+            next_classes.extend(YamlDeclarativeSource._get_next_expand_classes(field.type))
+
+        for next_class in next_classes:
+            YamlDeclarativeSource.expand_schema_interfaces(next_class, visited)
+        return copy_cls
+
+    @staticmethod
+    def _get_next_expand_classes(field_type) -> list[type]:
+        """
+        Parses through a given field type and assembles a list of all underlying declarative components. For a concrete declarative class
+        it will return itself. For a declarative interface it will return its subclasses. For declarative components in a generic type
+        it will return the unpacked classes. Any non-declarative types will be skipped.
+        :param field_type: A field type that
+        :return:
+        """
+        generic_type = typing.get_origin(field_type)
+        if generic_type is None:
+            module = field_type.__module__
+            # We can only continue parsing declarative components since we explicitly inherit from the JsonSchemaMixin class which is
+            # used to generate the final json schema
+            if "airbyte_cdk.sources.declarative" in module and not isinstance(field_type, EnumMeta):
+                subclasses = field_type.__subclasses__()
+                if subclasses:
+                    return subclasses
+                else:
+                    return [field_type]
+        elif generic_type == list or generic_type == Union:
+            next_classes = []
+            for underlying_type in typing.get_args(field_type):
+                next_classes.extend(YamlDeclarativeSource._get_next_expand_classes(underlying_type))
+            return next_classes
+        return []
+
+
+class SchemaEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, property):
+            return str(obj)
+        elif isinstance(obj, HttpMethod):
```

Comment:
is this because `HttpMethod` is an enum? Is `RequestOptionType` somehow not a problem?

Reply:
Yep this avoids the serialization issue for the `HttpMethod` enum. The reason why we need it for `HttpMethod` is because we assign the enum value `HttpMethod.GET` as the default so the enum (not just a string) shows up in the schema [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py#L44).

But for `RequestOptionType` since we don't assign the enum anywhere, enum class shows up in the schema, but not the enum value. That's why we didn't run into the serialization problem

Reply:
can we check if the type is an enum instead so we don't have to add another case to this method when we add another enum with a default value?

Reply:
good point, will change this